packages/cliqz-component-ui-movie-showtimes/src/styles.ts

```diff
@@ -0,0 +1,29 @@
+import { ViewStyle } from 'react-native';
+
+type WebStyles = {
+  cursor?: 'pointer' | 'default';
+};
+
+export type CliqzViewStyle = ViewStyle & WebStyles;
+
+export function getStyle<T>(baseStyles: T, styleOverwrites: Partial<T> | undefined): T {
+  if (styleOverwrites) {
+    for (const key of Object.keys(styleOverwrites)) {
+      // @ts-ignore
+      baseStyles[key] = { ...baseStyles[key], ...styleOverwrites[key] };
+    }
+  }
+
+  return baseStyles;
+}
```

Comment:
@chrmod @mai-cliqz I think I found a way to do it completely typed (without the `@ts-ignore`):

```typescript
type UpdateSpec<T> = {[P in keyof T]?: T[P] & UpdateSpec<T[P]>};

function getStyle<T>(base: T, partial: Partial<UpdateSpec<T>> = {}): T {
  const result: T = {...base};

  for (const key of (Object.keys(partial) as (Array<keyof T>))) {
    const value: T[keyof T] | undefined = partial[key];
    if (value !== undefined) {
      result[key] = { ...result[key], ...value };
    }
  }

  return result;
}
```

Reply:
@remusao Very nice one. It works

Reply:
@mai-cliqz BTW, I think these two functions should probably be extracted into an `utils` package or something: they are pretty generic and can be reused for any type (not only styles)